### DISCUSSION TITLE
How to have the client at startup and always running

### DISCUSSION THREAD

**Discussion by johnsidoli** [abraunegg/onedrive#1775] [Q&A] [answered]
Hi just installed client, looking good and working well with default config. Thanks for developing this, it's brilliant.

Meanwhile, I've got my head around most of it... next step is I want to have it always running in the background on LInux Mint.

Am I correct these are the instructions to do so? https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#onedrive-service-running-as-root-user-via-systemd-arch-ubuntu-debian-opensuse-fedora

I am not 100% sure if 'system service' is the technical term for 'running all the time' so didn't want to just follow it blindly.

Also... do I want to run it as root or as my own user? I only have root and my user on the computer, and always login to my user (unless I am running sudo for system install/changes/etc)... which should I do? Sorry if I am being ignorant here.

Finally... is there any way I can have notifications of big / certain events whilst it runs in the background? Like, notify me of all deleted files? I am paranoid of something going on in the background and not realising it deleted a bunch of stuff until too late... So some form of system notification that tells you when stuff is deleted etc, would be really good... i.e. aside of the logging functionality... which is great, but it produces streams and streams and lines and lines of text which is not easily digestible just by opening the file (and I don't want to have to read the log every day to check about deleted data). Pop up system notifications for example... that would be brilliant.

**Comment by abraunegg** [comment, marked as answer]
@johnsidoli 
> I am not 100% sure if 'system service' is the technical term for 'running all the time' so didn't want to just follow it blindly.

This is the correct process - to run all the time, it is a system service, and on Linux, this is done via either systemd, initd or runit - it depends on the distribution, but mostly systemd is correct.

> Also... do I want to run it as root or as my own user? I only have root and my user on the computer, and always login to my user (unless I am running sudo for system install/changes/etc)... which should I do? Sorry if I am being ignorant here.

Generally - you should not be using 'root' for anything other than installing software or other tasks that need elevated permissions. Run the systemd service as your normal user account.

> Finally... is there any way I can have notifications of big / certain events whilst it runs in the background? Like, notify me of all deleted files? I am paranoid of something going on in the background and not realising it deleted a bunch of stuff until too late...

Notification support is built in (see: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#notifications) - but, depends on how the client as been built. If you have installed from a package - it should already be enabled. If you have compiled from source, you need to have enabled that option at compile time: https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#build-options

**Comment by johnsidoli** [comment]
Thank you!!! <3
